python/lsst/pipe/tasks/isolatedStarAssociation.py

```diff
@@ -0,0 +1,595 @@
+                                      np.deg2rad(table[self.config.dec_column]))
+
+            tables.append(table[tract_use])
+
+        # Combine tables
+        star_obs_cat = np.concatenate(tables)
+
+        return star_obs_cat
+
+    def _get_source_table_visit_columns(self):
+        """Get the list of sourceTable_visit columns from the config.
+
+        Returns
+        -------
+        all_columns : `list` [`str`]
+            All columns to read
+        persist_columns : `list` [`str`]
+            Columns to persist (excluding selection columns)
+        """
+        columns = [self.config.id_column,
+                   'visit', 'detector',
+                   self.config.ra_column, self.config.dec_column,
+                   self.config.physical_filter_column, self.config.band_column,
+                   self.config.inst_flux_field, self.config.inst_flux_field + 'Err']
+        columns.extend(self.config.extra_columns)
+
+        all_columns = columns.copy()
+        if self.source_selector.config.doFlags:
+            all_columns.extend(self.source_selector.config.flags.bad)
+        if self.source_selector.config.doUnresolved:
+            all_columns.append(self.source_selector.config.unresolved.name)
+        if self.source_selector.config.doIsolated:
+            all_columns.append(self.source_selector.config.isolated.parentName)
+            all_columns.append(self.source_selector.config.isolated.nChildName)
+
+        return all_columns, columns
+
+    def _match_primary_stars(self, primary_bands, star_obs_cat):
+        """Match primary stars.
+
+        Parameters
+        ----------
+        primary_bands : `list` [`str`]
+            Ordered list of primary bands.
+        star_obs_cat : `np.ndarray`
+            Catalog of star observations.
+
+        Returns
+        -------
+        primary_star_cat : `np.ndarray`
+            Catalog of primary star positions
+        """
+        ra_col = self.config.ra_column
+        dec_col = self.config.dec_column
+
+        dtype = self._get_primary_dtype(primary_bands)
+
+        primary_star_cat = None
+        for primary_band in primary_bands:
+            use = (star_obs_cat['band'] == primary_band)
+
+            ra = star_obs_cat[ra_col][use]
+            dec = star_obs_cat[dec_col][use]
+
+            with Matcher(ra, dec) as matcher:
+                idx = matcher.query_self(self.config.match_radius/3600., min_match=1)
+
+            count = len(idx)
+
+            if count == 0:
+                self.log.info('Found 0 primary stars in %s band.', primary_band)
+                continue
+
+            band_cat = np.zeros(count, dtype=dtype)
+            band_cat['primary_band'] = primary_band
+
+            # rotate if necessary
+            rotated = False
+            if ra.min() < 60.0 and ra.max() > 300.0:
+                ra_temp = (ra + 180.0) % 360. - 180.
+                rotated = True
+            else:
+                ra_temp = ra
```

Comment:
I'll add some more comments (and rename rotated to remapped), but here's what's going on.  The inputs should not be all sky, I'm assuming they're less than 240 degrees across (seems reasonable for something run per tract...).  And this is just a quick check if the coords are anywhere close to 0 degrees in RA.  And then the little rotation trick with add, modulo, subtract is a clever way of taking any set of points and moving them to the `[-180 <= ra < 180]` domain.  This way when I take mean positions below I don't have to worry about stars that have observations on both sides of 0/360.  Of course, this would mess up things that are on both sides of +/-180 so that's why I only do the remapping if it isn't a tract that's near 180.

Reply:
Gotcha, that makes sense.